### PR TITLE
Refactor wkcuber for Zarr

### LINKED ISSUE #580: [wkcuber] cleanup testing
Currently the test setup consists of different scripts as well as pytest tests. The following commands are run in CI:
```bash
tar -xzvf testdata/WT1_wkw.tar.gz
poetry run pytest tests
poetry run tests/scripts/all_tests.sh
```

There's also a `test.sh` which is outdated atm. We should ensure that this script works, is used in CI and reflects all tests needed and expected. Additionally, we might double-check some of the scripts, they might also be refactored to pytest tests.

Related: #409

### LINKED ISSUE #654: Enable use of Zarr datasets in wkcuber
The PR #627 brings initial Zarr support to the Dataset API. However, it would be cool to also be able to use the `wkcuber` utilities to read/write Zarr datasets. Since the `wkcuber` largely builds upon the functionality of the Dataset API, in many cases Zarr support should be as easy as adding a `--data_format zarr` CLI arg. In other cases, e.g. image conversion or metadata guessing, there may be some challenges.
Additionally, I think the new `chunk_size` and `chunks_per_shard` parameters should also be exposed via the `wkcuber` CLI args.

### PR DESCRIPTION
**Pull request by normanrz**
### Description:
- This PR enables the use of Zarr datasets in wkcuber. Most commands now have the `--data_format`, `--chunks_per_shard`, and `--chunk_size` flags.
- The flag `--wkw_file_len` is deprecated (use `--chunks_per_shard` instead).
- S3 and HTTP support is enabled by environment variables.
- Other wkcuber refactorings
  - `wkcuber.recubing` is refactored to be a thin wrapper around `Dataset.copy_dataset`
  - `wkcuber.export_wkw_as_tiff` is refactored to use the Dataset-API internally.
  - `wkcuber.api` is marked as deprecated.
  - `wkcuber.utils` is also marked as deprecated. This probably needs fixing in voxelytics.
  - `wkcuber.downsampling_utils` and `wkcuber.upsampling_utils` are removed. They have been deprecated for a while.
  - There is now a new `wkcuber._internal` package that hosts `image_readers`, `knossos`, `utils`, and `versioning`.
- The CLI tests of wkcuber are now run through pytest in `test_cli.py`.

### Issues:
- fixes #654 
- fixes #580

### Todos:
Make sure to delete unnecessary points or to check all before merging:
 - [x] Updated Changelog

### CODE REVIEW

**Review by jstriebel — COMMENTED** [review]
Awesome stuff, thanks a ton for the cleanup und structuring! Mostly LGTM already :+1:, I still need to review four larger files in detail, just leaving some minor comments:

File: wkcuber/wkcuber/check_equality.py [hunk 1]
```diff
@@ -86,62 +55,54 @@ def check_equality(
 
     logging.info(f"Comparing {source_path} with {target_path}")
 
-    wk_src_dataset = Dataset.open(source_path)
-    src_layer_names = list(wk_src_dataset.layers.keys())
-    target_layer_names = [
-        layer["name"] for layer in detect_layers(target_path, 0, False)
-    ]
-    assert set(src_layer_names) == set(
```
[the remark this answers is not in the document]
Comment (normanrz) [reply wkcuber/wkcuber/check_equality.py:94]: It is checked in the `else` branch below (lines 76-78). When a user supplies a layer name via the CLI arg, only that layer should be checked. Whether the other layers exists shouldn't matter.

File: wkcuber/wkcuber/__init__.py [hunk 2]
```diff
@@ -5,9 +5,14 @@
...
+
 from .api.dataset import Dataset
+from .compress import compress_mag
 from .cubing import cubing
 from .downsampling import downsample_mags
-from .compress import compress_mag
-from .mag import Mag
 from .metadata import write_webknossos_metadata
+
+warnings.filterwarnings("once", category=DeprecationWarning)
```
Comment (jstriebel) [inline wkcuber/wkcuber/__init__.py:18]: I guess that's the default anyways, or is it configured differently somewhere else?

**Review by jstriebel — APPROVED** [review]
LGTM 👍

File: wkcuber/wkcuber/convert_knossos.py [hunk 3]
```diff
@@ -49,64 +51,93 @@ def create_parser() -> ArgumentParser:
...
+        knossos_cubes = np.array(list(source_knossos.list_cubes()))
+        if len(knossos_cubes) == 0:
+            logging.error(
+                "No input KNOSSOS cubes found. Make sure to pass the path which points to a KNOSSOS magnification (e.g., testdata/knossos/color/1)."
             )
+            exit(1)
+
+        min_xyz = knossos_cubes.min(axis=0) * CUBE_EDGE_LEN
+        max_xyz = (knossos_cubes.max(axis=0) + 1) * CUBE_EDGE_LEN
+        print(min_xyz, max_xyz)
```
Comment (jstriebel) [inline wkcuber/wkcuber/convert_knossos.py:125]: Please use the logging module instead

File: wkcuber/wkcuber/__init__.py (hunk 2 above)
Comment (normanrz) [reply wkcuber/wkcuber/__init__.py]: Yes, probably.

**Pull request merged**